### PR TITLE
[REG-2145] Add test APIs for using bot segments, actions, and criteria

### PR DESCRIPTION
**Pull request by RG-nAmKcAz**
- Adds APIs to our RGTestUtils to allow users to think about their Unity tests in terms of our BotActions and BotCriteria
  - Allows running a bot segment or bot segment list (in addition to the existing sequence support we had)
  - (I put these in to unlock the use of CV or other powerful criteria types in regular unity tests)
    - Allows running a bot action as a standalone test step
    - Allows waiting for bot criteria as a standalone test step
- Adds test with actions and criteria (this was hard to get right with the setup needed based on which input system and which event system was in use)
- Moves some test code to a more appropriate location 
- Hardens some code around inputData to make this easier and allow null/non-set fields instead of requiring empty objects 
- TODO: Will need to create a follow up task to document these things for both legacy and new input/event systems

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

File: src/gg.regression.unity.bots/Tests/TestFramework/Scripts/RGTestUtils.cs [hunk 1]
```diff
@@ -1,29 +1,36 @@
...
 {
-    /// <summary>
-    /// Utilities for running tests within the Unity Test Runner using Regression Games features.
-    /// </summary>
+    /**
+     * <summary>
+     * Utilities for running tests within the Unity Test Runner using Regression Games features.
+     * </summary>
+     */
     public class RGTestUtils
```
Comment (RG-nAmKcAz) [inline src/gg.regression.unity.bots/Tests/TestFramework/Scripts/RGTestUtils.cs:25]: Github folds this file by default, but this file is where the bulk of the meaningful changes are

File: src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/KeyFrameCriteria.cs [hunk 2]
```diff
@@ -1,11 +1,13 @@
 ﻿using System;
 using System.Text;
+using System.Threading;
```
Comment (vontell) [inline src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/KeyFrameCriteria.cs:3]: Was this added accidentally? I don't see any change here that would need this import but maybe I am missing something.

**Review by vontell — APPROVED** [review]
This is awesome! I've been hoping for something like this for a while so great to see this get in here. Thanks for adding the tests too, serves as a good example.

One small question about an import, so won't block on that, but am curious about that

File: src/gg.regression.unity.bots/Tests/Scripts/BotSegments/BotSegmentsTests.cs [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+            RGUtils.SetupOverrideEventSystem();
+
+            GameObject eventSystem = GameObject.Find("EventSystem");
+            var eventSys = eventSystem.GetComponent<EventSystem>();
+            RGLegacyInputWrapper.UpdateMode = RGLegacyInputUpdateMode.AUTOMATIC;
+            RGLegacyInputWrapper.StartSimulation(eventSys);
+        }
+
+        [UnityTest]
+        public IEnumerator TestRGUtilsBotActionsAndCriteria()
```
Comment (RG-nAmKcAz) [inline src/gg.regression.unity.bots/Tests/Scripts/BotSegments/BotSegmentsTests.cs:48]: This is the new sample test using these APIs

**Review by addisonbgross — APPROVED** [review]

**Pull request merged**